tools/isobuild/bundler.js

```diff
@@ -491,7 +491,13 @@ export class NodeModulesDirectory {
         if (fileStatus && fileStatus.isSymbolicLink()) {
           // If node_modules/.bin/command is a symlink, determine the
           // answer by calling isWithinProdPackage(real).
-          return isWithinProdPackage(files.realpathOrNull(path));
+          // guard against broken symlinks (#11241)
+          const realpath = files.realpathOrNull(path);
+          if (realpath) {
```

Comment:
I would change this to 
```suggestion
          if (!realpath) {
              throw new Error(`Broken symbolic link encountered at ${path}`);
          }
          return isWithinProdPackage(realpath);
```

so we avoid if/else and also handle the problematic case first.